examples/draw-features.js

```diff
@@ -29,9 +29,17 @@ let draw; // global so we can remove it later
 function addInteraction() {
   const value = typeSelect.value;
   if (value !== 'None') {
+    let maxPoints;
+    if (value === 'MultiPoint') {
+      // By default MultiPoint drawing casts a single Point
+      // as a MultiPoint geometry. Override this to allow
+      // the user to choose when to finish drawing.
+      maxPoints = Infinity;
```

Comment:
In a real life use case, I think drawing an infinite number of points in a single MultiPoint geometry is not very common. A typical UI will use a `finishCondition` or a button connected with a `finishDrawing()` call when multipoint drawing is done.

Reply:
Setting Infinity should make it behave like a drawing a LineString.  2 clicks on the same point will finish it.  I've not tested with a `finishCondition` but it goes through the same code.

Reply:
The code of this interaction is becoming increasingly hard to understand. I know I approved the use of `LINE_STRING` mode for drawing circles. But now that `LINE_STRING` mode is also being used to draw multi points, I regret that. Thinking more about it now, I wish there was still a `CIRCLE` mode. Even if it is more or less just an alias for `LINE_STRING`, it could help avoid if-clauses like the ones that were introduced in https://github.com/openlayers/openlayers/pull/11753/files#diff-91329c8a038835b6a5e7bae5e13f15e7fa5499b3414f0e094a5bfe347efc5a32R957.

For multipoints, the code would also be easier to understand if the mode was still `POINT`. And I'm still not convinced that this pull request is serving a useful purpose. Because if I want multipoints, I can as well draw points and use the `drawend` event to take the point and add it to a multipoint geometry. Same as for other multi geometries.

Reply:
It can also be easily done using a geometry function  https://codesandbox.io/s/draw-features-forked-yhmv7?file=/main.js which does the same as the PR without adding anything extra to the interaction code, so if it has limited use I will close this.

Reply:
Thanks @mike-000. Please close this. If you'd be able to implement the code readability improvements I [suggested](https://github.com/openlayers/openlayers/pull/11775#discussion_r531200283), it would be much appreciated.